cloud-controller-manager/do/certificates_test.go

```diff
@@ -0,0 +1,318 @@
+				service, _ := createServiceAndCert(lb.ID, "meow", certTypeLetsEncrypt)
+				return service
+			},
+			expectedServiceCertID: "test-cert-id",
+			expectedLBCertID:      "test-cert-id",
+		},
+		{
+			name: "[lets_encrypt] LB cert ID does not exit and service cert ID does",
+			setupFn: func(lbService fakeLBService, certService kvCertService) *v1.Service {
+				lb, _ := createHTTPSLB(443, 30000, "test-lb-id", "test-cert-id", certTypeLetsEncrypt)
+				lbService.store[lb.ID] = lb
+
+				service, cert := createServiceAndCert(lb.ID, "service-cert-id", certTypeLetsEncrypt)
+				certService.store[cert.ID] = cert
+				return service
+			},
+			expectedServiceCertID: "service-cert-id",
+			expectedLBCertID:      "service-cert-id",
+		},
+
+		// custom test cases
+		{
+			name: "[custom] LB cert ID and service cert ID match ",
+			setupFn: func(lbService fakeLBService, certService kvCertService) *v1.Service {
+				lb, cert := createHTTPSLB(443, 30000, "test-lb-id", "test-cert-id", certTypeCustom)
+				lbService.store[lb.ID] = lb
+				certService.store[cert.ID] = cert
+				return createServiceWithCert(lb.ID, cert.ID)
+			},
+			expectedServiceCertID: "test-cert-id",
+			expectedLBCertID:      "test-cert-id",
+		},
+		{
+			name: "[custom] LB cert ID and service cert ID differ and both certs exist",
+			setupFn: func(lbService fakeLBService, certService kvCertService) *v1.Service {
+				lb, cert := createHTTPSLB(443, 30000, "test-lb-id", "test-cert-id", certTypeCustom)
+				lbService.store[lb.ID] = lb
+				certService.store[cert.ID] = cert
+
+				service, cert := createServiceAndCert(lb.ID, "service-cert-id", certTypeCustom)
+				certService.store[cert.ID] = cert
+				return service
+			},
+			expectedServiceCertID: "service-cert-id",
+			expectedLBCertID:      "service-cert-id",
+		},
+		{
+			name: "[custom] LB cert ID does not exit and service cert ID does",
+			setupFn: func(lbService fakeLBService, certService kvCertService) *v1.Service {
+				lb, _ := createHTTPSLB(443, 30000, "test-lb-id", "test-cert-id", certTypeCustom)
+				lbService.store[lb.ID] = lb
+
+				service, cert := createServiceAndCert(lb.ID, "service-cert-id", certTypeCustom)
+				certService.store[cert.ID] = cert
+				return service
+			},
+			expectedServiceCertID: "service-cert-id",
+			expectedLBCertID:      "service-cert-id",
+		},
+	}
+
+	nodes := []*v1.Node{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-1",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-2",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-3",
+			},
+		},
+	}
+	droplets := []godo.Droplet{
+		{
+			ID:   100,
+			Name: "node-1",
+		},
+		{
+			ID:   101,
+			Name: "node-2",
+		},
+		{
+			ID:   102,
+			Name: "node-3",
+		},
+	}
+	fakeDroplet := fakeDropletService{
+		listFunc: func(context.Context, *godo.ListOptions) ([]godo.Droplet, *godo.Response, error) {
+			return droplets, newFakeOKResponse(), nil
+		},
+	}
+
+	for _, tc := range testcases {
```

Comment:
refactor: `s/tc/test/` to be consistent with the naming style we employ or recent tests.